src/main/webapp/resources/js/apis/metadata/field.ts

```diff
@@ -25,11 +34,14 @@ export const fieldsApi = createApi({
         url: "",
         params: { projectId },
       }),
-      provides: (result) => [
-        ...result.map(({ id }) => ({ type: "MetadataFields", id })),
+      providesTags: (result) => [
```

Comment:
I think it's only been enforced in associated-projects.ts so far, so I'm just going to leave it for now.

Reply:
Yes, I just learned how to do this recently so I was just trying to point out to you that it is possible to type these and how it can be done.  Again, just trying to pass on some recently learned knowledge.  The reason I keep pointing out the return types of functions is, obviously not for the client's benefit, but for us, and anyone who comes in the future to modify our code.  It is much easier to look at the typed (whether inferred or explicitly defined) return value to understand what you have to work with than it is to try to find the method on the server (or the network tab) and see what is being returned.